ledfx/dedupequeue.py

```diff
@@ -0,0 +1,38 @@
+import asyncio
+import logging
+
+from ledfx.events import Event
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VisDeduplicateQ(asyncio.Queue):
+    def __init__(self, maxsize=0):
+        super().__init__(maxsize)
+
+    def put_nowait(self, item):
+        # Check if is a visualisation update message
+        if (
+            item.get("event_type") == Event.DEVICE_UPDATE
+            or item.get("event_type") == Event.VISUALISATION_UPDATE
+        ):
+            # check if it is a duplicate and just return without queing if it is
+            if any(
+                self.is_similar(item, existing_item)
+                for existing_item in self._queue
```

Comment:
this is the only code of concern. _queue is private, so relying on it can be dangerous if it ever mutates in the future, but should be more than stable enough for our needs, and if it mutates, then lets fix it then...